python/lsst/ts/ess/csc/data_client/siglent_ssa3000x_spectrum_analyzer_data_client.py

```diff
@@ -0,0 +1,221 @@
+# This file is part of ts_ess_csc.
+#
+# Developed for the Vera C. Rubin Observatory Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["SiglentSsa3000xSpectrumAnalyzerDataClient"]
+
+import asyncio
+import logging
+import types
+from datetime import datetime, timedelta
+from typing import Any
+
+import numpy as np
+import yaml
+from lsst.ts import salobj, tcpip
+from lsst.ts.ess import common
+from lsst.ts.utils import current_tai, make_done_future
+
+# The maximum number of timeouts to allow before raising a TimeoutError.
+MAX_ALLOWED_READ_TIMEOUTS = 5
+
+# The amount of time to sleep between sending commands (seconds).
+SLEEP_TIME = 1
```

Comment:
I am pretty sure this not used -- that you use the similar config parameter instead.